src/snowflake/snowpark/_internal/udf_utils.py

```diff
@@ -165,24 +165,20 @@ def extract_return_type_from_udtf_type_hints(
     elif return_type_hint is None:
         return None
     else:
-        # Vectorized UDTF
-        if not installed_pandas:
-            raise RuntimeError(
-                "No pandas detected in local environment, please install pandas to use vectorized UDTF"
-            )
-        elif typing.get_origin(return_type_hint) == PandasDataFrame:
-            return PandasDataFrameType(
-                col_types=[
-                    python_type_to_snow_type(x)[0]
-                    for x in typing.get_args(return_type_hint)
-                ],
-                col_names=output_schema,
-            )
-        elif return_type_hint is pandas.DataFrame:
-            return PandasDataFrameType(
-                []
-            )  # placeholder, indicating the return type is pandas DataFrame
-        elif return_type_hint is NoneType:
+        if installed_pandas:  # Vectorized UDTF
+            if typing.get_origin(return_type_hint) == PandasDataFrame:
```

Comment:
This can be a follow up and I'm not sure if it is feasible, but I wonder if we should introduce something like `snowflake.connector.options` where we have the import handled in one place 🤔

Reply:
Yea I agree, there needs to be a cleaner and more robust way of doing this.

Reply:
do you import `PandasDataFrameType` or `PandasSeriesType` from somewhere? If pandas is not installed, how do the import work?

Reply:
👍  https://snowflakecomputing.atlassian.net/browse/SNOW-881404